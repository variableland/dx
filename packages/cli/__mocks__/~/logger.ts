import type { AnyLogger } from "~/logger";

const StubLogger: AnyLogger = {
  info: () => {},
  error: () => {},
  debug: () => {},
  subdebug: () => () => {},
};

export { StubLogger as Log };
