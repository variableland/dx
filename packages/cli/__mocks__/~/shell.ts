import { vi } from "vitest";

export const $ = vi.fn(function fakeShell(strs: string[], ...args: any) {
  let output = "";
  let argsIndex = 0;

  const stringifyArg = (arg: any) => (typeof arg.stdout === "string" ? arg.stdout : arg);

  strs.forEach((str) => {
    if (str === "") {
      if (args[argsIndex]) {
        output += stringifyArg(args[argsIndex]);
        argsIndex += 1;
      }
    } else {
      output += str;
    }
  });

  return output;
});
