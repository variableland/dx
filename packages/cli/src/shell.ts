import { $ as make$ } from "zx";
import { quote } from "./utils/quote";

export const $ = make$({
  verbose: true,
  quote: (arg: any) => (typeof arg.stdout !== "undefined" ? arg.stdout : quote(arg)),
});
