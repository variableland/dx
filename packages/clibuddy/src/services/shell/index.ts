export * from "./create";
export * from "./shell";
export * from "./types";
export * from "./utils";
