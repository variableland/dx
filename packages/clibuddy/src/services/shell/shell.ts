import { type Options, type Shell, $ as make$ } from "zx";

type ShellOptions = Partial<Options>;

export class ShellService {
  #shell: Shell;
  #quiet: Shell;

  constructor(options: ShellOptions) {
    this.#shell = make$(options);
    this.#quiet = make$({ ...options, verbose: false });
  }

  get $() {
    return this.#shell;
  }

  get $quiet() {
    return this.#quiet;
  }
}
