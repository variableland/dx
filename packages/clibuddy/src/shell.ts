import fs from "node:fs";
import { $ as make$ } from "zx";

export const isRaw = (arg: unknown): arg is { stdout: string } =>
  typeof arg === "object" && arg !== null && "stdout" in arg && typeof arg.stdout === "string";

export const cwd = fs.realpathSync(process.env.PWD ?? process.cwd());

// Inspired by https://dub.sh/6tiHVgn
export function quote(arg: string) {
  if (/^[\w./:=@-]+$/i.test(arg) || arg === "") {
    return arg;
  }

  return arg
    .replace(/\\/g, "\\\\")
    .replace(/'/g, "\\'")
    .replace(/\f/g, "\\f")
    .replace(/\n/g, "\\n")
    .replace(/\r/g, "\\r")
    .replace(/\t/g, "\\t")
    .replace(/\v/g, "\\v")
    .replace(/\0/g, "\\0");
}

export const $ = make$({
  cwd,
  verbose: true,
  quote: (arg: unknown) => {
    if (typeof arg === "string") {
      return quote(arg);
    }

    if (isRaw(arg)) {
      return arg.stdout;
    }

    throw TypeError(`Unsupported argument type: ${typeof arg}`);
  },
});
