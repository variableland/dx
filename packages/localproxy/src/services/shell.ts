import { createShellService } from "@vlandoss/clibuddy";

export const quietShell = createShellService({
  quiet: true,
  verbose: false,
});

export const silentShell = quietShell.child({
  stdio: ["ignore", "ignore", "ignore"],
});

export const verboseShell = quietShell.child({
  quiet: false,
  verbose: true,
});
