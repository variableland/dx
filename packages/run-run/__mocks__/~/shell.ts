import { vi } from "vitest";
import { isRaw } from "../../src/utils/is-raw";

export const $ = vi.fn(function fakeShell(strs: string[], ...args: string[]) {
  let output = "";
  let argsIndex = 0;

  const stringifyArg = (arg: unknown) => (isRaw(arg) ? arg.stdout : arg);

  for (const str of strs) {
    if (str === "") {
      if (args[argsIndex]) {
        output += stringifyArg(args[argsIndex]);
        argsIndex += 1;
      }
    } else {
      output += str;
    }
  }

  return output;
});
