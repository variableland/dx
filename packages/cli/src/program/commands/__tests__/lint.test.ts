import { expect, test, vi } from "vitest";
import { parseProgram, execCli } from "test/helpers";
import { $ } from "~/shell";

test("should display help message", async () => {
  const { stdout } = await execCli("lint --help");

  expect(stdout).toMatchSnapshot();
});

test("should run lint command", async () => {
  await parseProgram(["lint"]);

  expect(vi.mocked($)).toBeCalledTimes(1);
  expect(vi.mocked($).mock.results[0]?.value).toMatchInlineSnapshot(
    `"biome lint --no-errors-on-unmatched --colors=force"`,
  );
});

test("should run lint command with --fix flag", async () => {
  await parseProgram(["lint", "--fix"]);

  expect(vi.mocked($)).toBeCalledTimes(1);

  expect(vi.mocked($).mock.results[0]?.value).toMatchInlineSnapshot(
    `"biome lint --no-errors-on-unmatched --colors=force --fix"`,
  );
});
