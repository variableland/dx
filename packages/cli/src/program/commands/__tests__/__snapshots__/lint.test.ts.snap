// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should display help message 1`] = `
"Usage: run-run lint [options]

lint the code ðŸ§¹

Options:
  -f, --fix   fix the linting issues
  -h, --help  display help for command

Under the hood, this command uses the \`eslint\` CLI to lint the code.
"
`;
